test/test_data/test_feature_validator.py

```diff
@@ -439,3 +457,23 @@ def test_no_new_category_after_fit():
     validator.fit(x)
     x['A'] = x['A'].apply(lambda x: x*x)
     validator.transform(x)
+
+
+def test_unknown_encode_value():
+    x = pd.DataFrame([
+        {'a': -41, 'b': -3, 'c': 'a', 'd': -987.2},
+        {'a': -21, 'b': -3, 'c': 'a', 'd': -9.2},
+        {'a': 0, 'b': -4, 'c': 'b', 'd': -97.2},
+        {'a': -51, 'b': -3, 'c': 'a', 'd': 987.2},
+        {'a': 500, 'b': -3, 'c': 'a', 'd': -92},
+    ])
+    x['c'] = x['c'].astype('category')
+    validator = FeatureValidator()
+
+    # Make sure that this value is honored
+    validator.fit(x)
+    x['c'].cat.add_categories(['NA'], inplace=True)
+    x.loc[0, 'c'] = 'NA'  # unknown value
+    x_t = validator.transform(x)
+    # The first row should have a -1 as we added a new categorical there
+    assert -1 in x_t[0].tolist()
```

Comment:
Could you please check for the exact location of `-1` and also check that the other entries of that column/row are unchanged?

Reply:
Done!